catroid/src/org/catrobat/catroid/physics/PhysicsLook.java

```diff
@@ -128,10 +133,26 @@ public void setRotation(float degrees) {
 
 	@Override
 	public void setScale(float scaleX, float scaleY) {
+		Vector2 oldScales = new Vector2(getScaleX(), getScaleY());
+		if (scaleX < 0.0f || scaleY < 0.0f) {
+			scaleX = 0.0f;
+			scaleY = 0.0f;
+		}
+
+		int scaleXComp = Math.round(scaleX * SCALEFACTOR_ACCURACY);
+		int scaleYComp = Math.round(scaleY * SCALEFACTOR_ACCURACY);
+		if (scaleXComp == Math.round(oldScales.x * SCALEFACTOR_ACCURACY) && scaleYComp == Math.round(oldScales.y * SCALEFACTOR_ACCURACY)) {
+			return;
+		}
+
+		scaleX = scaleXComp / SCALEFACTOR_ACCURACY;
+		scaleY = scaleYComp / SCALEFACTOR_ACCURACY;
 		super.setScale(scaleX, scaleY);
```

Comment:
I don't know how the scaling for the physics stuff works and I don't have the time to dig into that right now. :crying_cat_face: 
But setScale should definitely call super.setScale with the original scaleX and scaleY values and without any guard clauses (no early returns). If and how you forward the scale changes to the physics engine is completely up to you. :wink: